packages/cuda/package.py

```diff
@@ -0,0 +1,661 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```

Comment:
Remind me, what's the reason for adding this to our repository, rather than using the one that comes with Spack? (Same applies to nvhpc.)

Reply:
The nvhpc isn't necessary if the latest stable release of spack is used (0.21.0) but the cuda one is necessary since spack-0.21.0 doesn't have cuda-12.2.2 yet. Also the README.md instructions on NESO still say to use spack-0.19.0. I've used spack-0.20.2 successfully, but given the fragility of NESO (and its dependencies) I was hesitant to try the latest version (I got to 0.20.2 after iteratively stepping up from 0.19.0). But I can double check if using spack-0.21.0 works. If it does then I can get rid of the nvhpc package.py

Reply:
Let's include these for now. We can consider deleting them later if we move to a more recent spack release. I think some changes will be needed to one of our other packages to make them work with spack 0.21.0.